src/vs/workbench/services/editor/browser/editorOverrideService.ts

```diff
@@ -625,6 +629,15 @@ export class EditorOverrideService extends Disposable implements IEditorOverride
 		}
 		return false;
 	}
+
+	private isDiffEditorInput(editor: UntypedEditorInput): editor is IResourceDiffEditorInput {
```

Comment:
Maybe this utility method could be exported from the place where `IResourceDiffEditorInput` is introduced, I think I have similar code in other places so there is chances of reuse.